### PR TITLE
core/services/ocr2: bump relay; set ContextValues.TransmitterID; cleanup

### PR DESCRIPTION
**Pull request by jmank88**
Requires:
- https://github.com/smartcontractkit/chainlink-relay/pull/131

### CODE REVIEW

File: core/services/ocr2/delegate.go [hunk 1]
```diff
@@ -294,26 +294,26 @@ func (d *Delegate) OnDeleteJob(jb job.Job, q pg.Queryer) error {
...
-		ContractID: spec.ContractID,
-		FeedID:     spec.FeedID,
+		ContractID:    spec.ContractID,
+		FeedID:        spec.FeedID,
+		TransmitterID: transmitterID,
 	}
-	lggr := logger.Sugared(d.lggr.Named("OCR2").With(ctxVals.Args()...))
-	feedID := spec.FeedID
-	if feedID != (common.Hash{}) {
-		lggr = logger.Sugared(lggr.With("feedID", spec.FeedID))
```
Comment (jmank88) [inline core/services/ocr2/delegate.go:315]: This is already set via `ContextValues.Args()`

File: core/services/ocr2/delegate.go [hunk 2]
```diff
@@ -335,6 +331,11 @@ func (d *Delegate) ServicesForSpec(jb job.Job) ([]job.ServiceCtx, error) {
 				spec.RelayConfig["sendingKeys"] = []string{transmitterID}
 			}
 
+			chain, err2 := d.chainSet.Get(big.NewInt(chainID))
+			if err2 != nil {
+				return nil, errors.Wrap(err2, "ServicesForSpec failed to get chainset")
+			}
```
Comment (jmank88) [inline core/services/ocr2/delegate.go:337]: Mercury only, so delay until necessary to unblock other products from using LOOP relayers.

File: core/services/ocr2/delegate.go [hunk 3]
```diff
@@ -348,7 +349,6 @@ func (d *Delegate) ServicesForSpec(jb job.Job) ([]job.ServiceCtx, error) {
 		}
 	}
 	spec.RelayConfig["effectiveTransmitterID"] = effectiveTransmitterID
-	lggr = logger.Sugared(lggr.With("transmitterID", transmitterID))
```
Comment (jmank88) [inline core/services/ocr2/delegate.go:351]: Moved to `ContextValues`

**Review by bolekk — DISMISSED** [review]
Nice! I'm closing https://github.com/smartcontractkit/chainlink/pull/9498

**Review by bolekk — APPROVED** [review]

**Pull request merged**